sfm/ui/notifications.py

```diff
@@ -11,13 +12,166 @@
 from django.contrib.sites.models import Site
 from django.core.urlresolvers import reverse
 
-
 from .models import User, CollectionSet, Collection, HarvestStat
 from .sched import next_run_time
 
 log = logging.getLogger(__name__)
 
 
+class MonitorSpace(object):
+    def __init__(self, volume_dir, threshold):
+        """
+        A class for monitor free space of the special directory
+        :param volume_dir: the volume mounted directory, considered as the id of the record.
+        :param threshold: the free space threshold.
+        :return:
+        """
+        # deal with the empty string
+        if not volume_dir:
+            volume_dir = 'None'
+        if not threshold:
+            threshold = '10GB'
+
+        self.space_msg_cache = {'volume_id': volume_dir, 'threshold': threshold, 'bar_color': 'progress-bar-success'}
```

Comment:
You don't need to make the change, but a slightly more pythonic way of handling the empty string would be:
```
self.space_msg_cache = {'volume_id': volume_dir or 'None', 'threshold': threshold or '10GB', 'bar_color': 'progress-bar-success'}
```